### PR TITLE
[Issue 1544] Fix [iOS] FilePicker.PickAsync never returns after picker is dismissed

### PR DESCRIPTION
**Pull request by BjornVanslembrouck**
### Description of Change ###

`FilePicker.PickAsync` never returns if the document picker controller is dismissed using a method other than tapping the Cancel button.

### Bugs Fixed ###

- Related to issue #1544 

### API Changes ###

None

### Behavioral Changes ###

The library now returns null when a user dismisses the picker by swiping down.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of `main` at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/FilePicker/FilePicker.ios.cs [hunk 1]
```diff
@@ -48,6 +48,14 @@ static Task<IEnumerable<FileResult>> PlatformPickAsync(PickOptions options, bool
                 }
             };
 
+            if (documentPicker.PresentationController != null)
+            {
+                documentPicker.PresentationController.Delegate = new PickerPresentationControllerDelegate
+                {
+                    PickHandler = urls => tcs.TrySetResult(Enumerable.Empty<FileResult>())
```
Comment (mattleibow) [inline Xamarin.Essentials/FilePicker/FilePicker.ios.cs:55]: Technically, we don't need to do what I am about to type, but maybe for consistency, we should extract the handler from `PickerDelegate.PickHandler ` and then call it here. Maybe a local method might be good?

**Comment by mattleibow** [comment]
I am not able to test the file picker selection. When I pick an image on iOS, nothing happens - no visual change and the picker stays up. I can cancel, and, `MediaPicker` works. I have an iPhone X... Am I holding it wrong?

**Comment by BjornVanslembrouck** [comment]
@mattleibow, not sure what is happening there. I can use all the options and pick them as expected
FilePicker_iPhoneXs

Or do you mean something else?

**Comment by mattleibow** [comment]
@BjornVanslembrouck I think it is an issue (somehow) due to running from Hot Restart on Windows. I booted my macOS up and now it works. So that was fun.

File: Xamarin.Essentials/FilePicker/FilePicker.ios.cs (hunk 1 above)
Comment (BjornVanslembrouck) [reply Xamarin.Essentials/FilePicker/FilePicker.ios.cs:55]: That is an option, however doing it this way is consistent with how it is already done in the `MediaPicker`. Both delegates have their own functionality which they are responsible for.

Comment (mattleibow) [reply Xamarin.Essentials/FilePicker/FilePicker.ios.cs:55]: Sounds good.

I am actually looking at the media picker API and I see it doesn't handle exceptions... I'll do a PR on that separately.

**Review by mattleibow — APPROVED** [review]

**Comment by mattleibow** [comment]
Thanks for jumping on this!

**Pull request merged**